machine_learning/vector_ops.hpp

```diff
@@ -0,0 +1,477 @@
+/**
+ * @file vector_ops.hpp
+ * @author [Deep Raval](https://github.com/imdeep2905)
+ * 
+ * @brief Various functions for vectors associated with [NeuralNetwork (aka Multilayer Perceptron)] 
+ * (https://en.wikipedia.org/wiki/Multilayer_perceptron).
+ * 
+ */
+#ifndef VECTOR_OPS_FOR_NN
+#define VECTOR_OPS_FOR_NN
+
+#include <iostream>
+#include <algorithm>
+#include <vector>
+#include <valarray>
+#include <chrono>
+#include <random>
+
+/**
```

Comment:
You may also want to add a namespace member (namespace inside namespace).
```suggestion
/**
 * @namespace machine_learning
 * @brief Machine Learning algorithms
 */
namespace machine_learning {
/**
```